mobile-app/app/screens/AppNavigator/screens/Dex/components/PoolPairCards/PoolPairCards.tsx

```diff
@@ -116,17 +116,29 @@ export function PoolPairCards ({
 
   useEffect(() => {
     setIsSearching(false)
-    setFilteredYourPairs(
-      yourPairs.filter((pair) =>
-        pair.data.displaySymbol
-          .toLowerCase()
-          .includes(debouncedSearchTerm.trim().toLowerCase())
-      ).sort((a, b) =>
-        availablePairs.findIndex(x => x.data.id === a.data.id) -
-        availablePairs.findIndex(x => x.data.id === b.data.id
+    if (showSearchInput) {
+      if (debouncedSearchTerm !== undefined && debouncedSearchTerm.trim().length > 0) {
+        setFilteredYourPairs(
+          yourPairs.filter((pair) =>
+            pair.data.displaySymbol
+              .toLowerCase()
+              .includes(debouncedSearchTerm.trim().toLowerCase())
+          ).sort((a, b) =>
+            availablePairs.findIndex(x => x.data.id === a.data.id) -
```

Comment:
aware that the filtering logic is repeated in `DexScreen`, but as for this PR sake, we could refactor the sorting logic into a function in the mean time

Reply:
ok